src/lightning_app/utilities/cli_helpers.py

```diff
@@ -214,3 +223,97 @@ def _arrow_time_callback(
             return arrow.get(value)
         except (ValueError, TypeError):
             raise click.ClickException(f"cannot parse time {value}")
+
+
+def _is_valid_release(release):
+    version, release = release
+    version = packaging.version.parse(version)
+    if any(r["yanked"] for r in release) or version.is_devrelease or version.is_prerelease:
+        return False
+    return True
+
+
+@functools.lru_cache(maxsize=1)
+def _is_old_version() -> bool:
+    if get_dist_path_if_editable_install(__package_name__):
+        # Always return False in editable mode
+        return False
+    try:
+        response = requests.get(f"https://pypi.org/pypi/{__package_name__}/json")
+        releases = response.json()["releases"]
+        if __version__ not in releases:
+            # Always return False if not installed from PyPI (e.g. dev versions)
+            return False
+        releases = {version: release for version, release in filter(_is_valid_release, releases.items())}
+        sorted_releases = sorted(
+            releases.items(), key=lambda release: release[1][0]["upload_time_iso_8601"], reverse=True
+        )
+        latest_version = sorted_releases[0][0]
+        return False if __version__ == latest_version else latest_version
+    except Exception:
+        # Return False if any exception occurs
+        return False
+
+
+def _redirect_command(executable: str):
+    """Redirect the current lightning CLI call to the given executable."""
+    subprocess.run(
+        [executable, "-m", "lightning"] + sys.argv[1:],
+        env=os.environ,
+    )
+
+    sys.exit()
+
+
+def _check_version_and_upgrade():
+    new_version = _is_old_version()
+    if new_version:
```

Comment:
does this ask always or only once for each new release?

Reply:
Currently it would ask every time

Reply:
We could maybe cache the response if they choose 'no' and maybe just warn them on future runs